sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/InsertIntoHiveTable.scala

```diff
@@ -169,7 +169,10 @@ case class InsertIntoHiveTable(
 
     // All partition column names in the format of "<column name 1>/<column name 2>/..."
     val partitionColumns = fileSinkConf.getTableInfo.getProperties.getProperty("partition_columns")
-    val partitionColumnNames = Option(partitionColumns).map(_.split("/")).getOrElse(Array.empty)
+    // As the keys of partition spec `partition` is always lowercase, we should also lowercase the
+    // partition column names of the table here.
+    val partitionColumnNames =
+      Option(partitionColumns).map(_.split("/").map(_.toLowerCase)).getOrElse(Array.empty)
```

Comment:
It sounds like this is from Hive side. Thus, it should be always Lower Case? Without these changes, I ran the build. It sounds like it works well.

Can we use ExternalCatalog to get the partition info? I can make a try to see if it still works.